src/executable.cpp

```diff
@@ -325,277 +313,134 @@ void Executable::add_symbol(const ExeSymbol &symbol, bool overwrite)
-{
-    if (m_verbose)
+    if (js.contains(s_symbolsSection))
     {
-        printf("Loading section info...\n");
+        if (m_verbose)
+        {
+            printf("Loading symbols section...\n");
+        }
+        std::vector<ExeSymbol> symbols;
+        js.at(s_symbolsSection).get_to(symbols);
+        add_symbols(symbols, overwrite_symbols);
     }
 
-    for (auto it = js.begin(); it != js.end(); ++it)
+    if (js.contains(s_sectionsSection))
     {
-        std::string name;
-        it->at("name").get_to(name);
+        if (m_verbose)
+        {
+            printf("Loading sections info...\n");
+        }
+        std::vector<ExeSectionInfo> sections;
+        js.at(s_sectionsSection).get_to(sections);
 
-        // Don't try and load an empty section.
-        if (!name.empty())
+        for (const auto &sectionInfo : sections)
         {
-            ExeSectionInfo *section = find_section(name);
+            if (sectionInfo.name.empty())
+                continue;
 
-            if (section == nullptr)
+            ExeSectionInfo *existingSection = find_section(sectionInfo.name);
+            if (!existingSection)
             {
                 if (m_verbose)
                 {
-                    printf("Tried to load section info for section not present in this binary!\n");
-                    printf("Section '%s' info was ignored.\n", name.c_str());
+                    printf("Section '%s' not found in binary\n", sectionInfo.name.c_str());
                 }
                 continue;
             }
 
-            std::string type;
-            it->at("type").get_to(type);
-
-            section->type = to_section_type(type.c_str());
-
-            if (section->type == ExeSectionType::Unknown && m_verbose)
-            {
-                printf("Incorrect type specified for section '%s'.\n", name.c_str());
-            }
-
-            auto it_address = it->find("address");
-            if (it_address != it->end())
-            {
-                it_address->get_to(section->address);
-            }
-            auto it_size = it->find("size");
-            if (it_size != it->end())
-            {
-                it_size->get_to(section->size);
-            }
+            existingSection->type = sectionInfo.type;
+            existingSection->address = sectionInfo.address;
+            existingSection->size = sectionInfo.size;
         }
     }
-}
-
-void Executable::dump_sections(nlohmann::json &js) const
-{
-    if (m_verbose)
-    {
-        printf("Saving section info...\n");
-    }
 
-    for (const ExeSectionInfo &section : m_sections)
+    if (js.contains(s_objectsSection))
     {
-        const char *type_str = to_string(section.type);
-        js.push_back({{"name", section.name}, {"type", type_str}, {"address", section.address}, {"size", section.size}});
+        if (m_verbose)
+        {
+            printf("Loading objects section...\n");
+        }
+        js.at(s_objectsSection).get_to(m_targetObjects);
```

Comment:
It looks likely previously it merged objects in this container when loading. I don't know why. Now it does not do that anymore. Is that ok?

Reply:
Yeah it was checking if an object with same name exists and skipping if yes. I added this check back, in an update_objects method similar to update_sections.